src/DotNetWorker/Broker/FunctionBroker.cs

```diff
@@ -70,19 +70,15 @@ public async Task<InvocationResponse> InvokeAsync(FunctionInvocation invocation)
                     var parameterBindings = executionContext.OutputBindings;
                     var result = executionContext.InvocationResult;
 
-                    foreach (var paramBinding in parameterBindings)
+                    // TODO: ParameterBinding shouldn't happen here
+                    foreach (var binding in parameterBindings)
                     {
-                        // TODO: ParameterBinding shouldn't happen here
-
-                        foreach (var binding in executionContext.OutputBindings)
+                        var parameterBinding = new ParameterBinding
```

Comment:
Seemed like we had nested loops and were re-adding same data (n times) in case of multiple bindings. Just want to confirm this was accidental. @brettsam, do you know?

Once I changed below to `ToRpcAsync` some of our E2E tests started failing as it seemed `GrpcHttpResponseType` was attempted to serialize multiple times. Nice to have tests that catch these :)

Reply:
Yes, completely accidental -- likely due to a refactor. Good catch!